DataStructure/Src/DS_2_StackAndQueue/DS_2_0_SqStack.cpp

```diff
@@ -120,7 +120,8 @@ int GetTopOther1(SqStack S) {
 void PrintStack(SqStack S) {
     printf("从栈顶元素开始，栈如下：\n");
     while (S.top >= 0) {//注意判空的条件
-        printf("S[%d]=%d\n", S.top, S.data[S.top--]);

```

Comment:
> The expressions in printf() will get evaluated first before print out, so the counter needs to be put outside the printf() function.

Thanks for your PR~
But the effect of `printf("S[%d]=%d\n", S.top, S.data[S.top--])` and ` printf("S[%d]=%d\n", S.top, S.data[S.top]);S.top--` is the same ~ 
Maybe this way of writing is a little hard to understand，their execution results are the same. 
Thanks~

Reply:
I can't agree. I thought they would produce the same results. But I compiled the code, the results are different. So I think in the printf() function, all expressions are getting evaluated before printed out, including S.top--. Could you try to compile this?

Reply:
Please take a note, in printf function, there is no guarantee as when S.top or S.data[S.top--] will get evaluated first. This is called a undefined behaviour, which means different compilers will probably get different results, even the same compiler will get different results in different times. 
Check this out: https://stackoverflow.com/questions/30452526/using-unary-increment-decrement-in-printf-function

Reply:
Thanks for your explanation, I don’t know the concept of undefined behavior, but new I got it ~ Thanks again！